cmd/object-api-common.go

```diff
@@ -276,8 +276,13 @@ func listObjects(ctx context.Context, obj ObjectLayer, bucket, prefix, marker, d
 	var eof bool
 	var nextMarker string
 
+	maxConcurrent := maxKeys / 10
```

Comment:
If max-keys is equal to 10000, then we will have 1000 concurrent calls, probably too much and it can go higher.

Reply:
Yeah, that will kill any disk/server.

Reply:
maxKeys is always truncated to 1000 @vadmeste